crates/yarn_slinger/tests/test_base/mod.rs

```diff
@@ -1,76 +1,328 @@
+                        Some(StepValue::String(command.0))
+                    );
+                })
+                .set_node_complete_handler(|_, _| {});
+        }
+        {
+            let test_plan = test_plan.clone();
+            dialogue.set_dialogue_complete_handler(move |_| {
+                let mut test_plan = test_plan.write().unwrap();
+                let Some(test_plan) = test_plan.as_mut() else {
+                    return;
+                };
+                test_plan.next();
+                assert_eq!(
+                    ExpectedStepType::Stop,
+                    test_plan.next_expected_step,
+                    "Stopped dialogue, but wasn't expecting to select it (was expecting {:?})",
+                    test_plan.next_expected_step
+                );
+            });
+        }
+
+        dialogue
+            .library_mut()
+            .register_function("assert", |value: YarnValue| {
+                let is_truthy: bool = value.try_into().unwrap();
+                assert!(is_truthy);
+                true
+            });
+        Self {
+            dialogue,
+            test_plan,
+            string_table,
+            runtime_errors_cause_panic,
         }
     }
 }
 
-pub trait TestCompilationJob {
-    fn from_test_source(source: &str) -> Self;
+impl TestBase {
+    /// Sets the current test plan to one loaded from a given path.
+    pub fn read_test_plan(self, path: impl AsRef<Path>) -> Self {
+        self.with_test_plan(TestPlan::read(path))
+    }
+
+    pub fn with_test_plan(self, test_plan: TestPlan) -> Self {
+        self.test_plan.write().unwrap().replace(test_plan);
+        self
+    }
+
+    pub fn with_runtime_failure_causes_no_panic(self) -> Self {
+        self.runtime_errors_cause_panic
+            .store(false, Ordering::Relaxed);
+        self
+    }
+
+    pub fn with_compilation(self, compilation: Compilation) -> Self {
+        let string_table = compilation
+            .string_table
+            .into_iter()
+            .map(|(k, v)| (LineId(k), v))
+            .collect();
+        self.with_program(compilation.program.unwrap())
+            .with_string_table(string_table)
+    }
+
+    pub fn with_program(mut self, program: Program) -> Self {
+        self.dialogue.add_program(program);
+        self
+    }
+
+    pub fn with_string_table(self, string_table: HashMap<LineId, StringInfo>) -> Self {
+        *self.string_table.write().unwrap() = string_table;
+        self
+    }
+
+    /// Executes the named node, and checks any assertions made during
+    /// execution. Fails the test if an assertion made in Yarn fails.
+    pub fn run_standard_testcase(&mut self) {
+        self.dialogue.set_node_to_start();
+
+        self.dialogue.continue_();
+        while self.dialogue.is_active() {
+            self.dialogue.continue_();
+        }
+    }
+
+    /// Returns the list of .node and.yarn files in the Tests/<directory> directory.
```

Comment:
is it really Tests (uppercase?)

Reply:
![image](https://github.com/yarn-slinger/yarn-slinger/assets/9047632/6aebf8e5-6560-47e8-b063-b979f3bc5cc5)
yep, it's from the Yarn Spinner git repo we include

Reply:
Maybe I should change it to `third-party/YarnSpinner/Tests`?

Reply:
Yeah, that would make it clearer :) But it's okay this way as well